### PR TITLE
update name to fewlines scoped

### PR DESCRIPTION
**Pull request by julien-leclercq**
## Details 

In order to publish this package on npm this pull request does the following changes:
- update the package name to explicitly scope it with `@fewlines`.
- add a `main` field in `package.json`
- add a `types` field in `package.json` 
- gather exports in an `index.ts` file 
- configure typescript to export type declarations

### CODE REVIEW

**Review by kdisneur — DISMISSED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  makePayload,
+  makeWorkflow
+} from "./workflow";
+
+export type Logger = Logger;
+export type Payload<Input> = Payload<Input>;
+export type UnexpectedError = UnexpectedError;
+export type Workflow<Input, Result> = Workflow<Input, Result>;
+
+module.exports = {
```
Comment (julien-leclercq) [inline src/index.ts]: this has been done this way to avoid redondant type annotation in this file.

Comment (Fenntasy) [reply src/index.ts]: I dont' understand the link between redondant type annotation and the use of `module.exports`? 🤔

Comment (julien-leclercq) [reply src/index.ts]: when I tried to use an `export` statement the compiler requested to add more type annotations.

Comment (Fenntasy) [reply src/index.ts]: And you saw it done like that in other places? I'm curious about the errors, I must admit.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
     "typescript"
   ],
   "license": "MIT",
-  "name": "ts-workflow",
+  "name": "@fewlines/ts-workflow",
+  "main": "dist/index.js",
+  "types": "index.d.ts",
```
Comment (kdisneur) [inline package.json]: Shouldn't we commit this file? Moreover the `declarationDir` is set to `./dist` so I don't understand why it's not in `./dist/index.d.ts` ?

File: src/index.ts [hunk 3]
```diff
@@ -3,16 +3,14 @@ import {
...
 export type Logger = Logger;
 export type Payload<Input> = Payload<Input>;
 export type UnexpectedError = UnexpectedError;
 export type Workflow<Input, Result> = Workflow<Input, Result>;
 
-module.exports = {
-  makePayload,
-  makeWorkflow
-};
+export const makePayload = workflowMakePayload;
```
Comment (Fenntasy) [inline src/index.ts]: Did you try something like `export { makePayload } from "./workflow";`?

File: package.json (hunk 2 above)
Comment (julien-leclercq) [reply package.json]: > Shouldn't we commit this file? 
nop this is published but not commit (if you want I can link guides I found that specified to do it this way) 

> Moreover the declarationDir is set to ./dist so I don't understand why it's not in ./dist/index.d.ts ?
Nice catch the path is wrong. 👍

**Review by kdisneur — DISMISSED** [review]

**Comment by kdisneur** [comment]
I guess you discussed the prefix `make` over `new`, `build`, etc.... That's the convention in JS / TS ?

**Comment by julien-leclercq** [comment]
> I guess you discussed the prefix `make` over `new`, `build`, etc.... That's the convention in JS / TS ?

Of course we did not :(

**Review by kdisneur — APPROVED** [review]

**Pull request merged**